promql/quantile.go

```diff
@@ -106,3 +106,33 @@ func quantile(q model.SampleValue, buckets buckets) float64 {
 	}
 	return bucketStart + (bucketEnd-bucketStart)*float64(rank/count)
 }
+
+// qauntile calculates the given quantile of a slice of floats.
+//
+// The slice will be sorted.
+// If 'buckets' has zero elements, NaN is returned.
+// If q<0, -Inf is returned.
+// If q>1, +Inf is returned.
+func quantile(q float64, values []float64) float64 {
+	if len(values) == 0 {
+		return math.NaN()
+	}
+	if q < 0 {
+		return math.Inf(-1)
+	}
+	if q > 1 {
+		return math.Inf(+1)
+	}
+	sort.Float64s(values)
```

Comment:
How about creating a `model.SampleValues` similar to `model.Samples` in the common/model package, which implements `sort.Interface`? It would be a bit of boilerplate code over there but would save a lot of float64 conversions here and create much more consistency.

While reading the code, my brain is wasting cycles all the time thinking why suddenly the SampleValues have to be floats.